src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugModuleFactory.cs

```diff
@@ -74,18 +76,13 @@ public class DebugModuleFactory : ModuleFactoryBase<IDebugRpcModule>
         _syncModeSelector = syncModeSelector ?? throw new ArgumentNullException(nameof(syncModeSelector));
         _badBlockStore = badBlockStore;
         _fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        _txProcessorSource = txProcessorSource;
         _logger = logManager.GetClassLogger();
     }
 
     public override IDebugRpcModule Create()
     {
-        ReadOnlyTxProcessingEnv txEnv = new(
-            _worldStateManager,
-            _blockTree,
-            _specProvider,
-            _logManager);
-
-        IReadOnlyTxProcessingScope scope = txEnv.Build(Keccak.EmptyTreeHash);
+        IReadOnlyTxProcessingScope scope = _txProcessorSource.Build(Keccak.EmptyTreeHash);
```

Comment:
Does the scope become disposed?

Reply:
Not here. The rpc module now become the scope.